tests/by-util/test_groups.rs

```diff
@@ -1,56 +1,172 @@
 use crate::common::util::*;
 
+// These tests run the GNU coreutils `(g)groups` binary in `$PATH` in order to gather reference values.
+// If the `(g)groups` in `$PATH` doesn't include a coreutils version string,
+// or the version is too low, the test is skipped.
+
+// The reference version is 8.32. Here 8.30 was chosen because right now there's no
+// ubuntu image for github action available with a higher version than 8.30.
+const VERSION_EXPECTED: &str = "8.30"; // Version expected for the reference `groups` in $PATH
+const VERSION_MULTIPLE_USERS: &str = "8.31";
```

Comment:
I am not a fan of hardcoding the version; it makes the maintenance much harder.
could you please try to find a better solution? thanks

Reply:
This is just for ease of use, so that tests automatically **skip** if the system they run on doesn't have a compatible version of `groups` in `$PATH`.
If there is GNU's coreutils `groups` in `$PATH` and if the Version is `>=` 
* `VERSION_MULTIPLE_USERS`: test `test_groups_username_multiple` will run
* `VERSION_EXPECTED`: tests `test_groups_username` and `test_groups` will run.

GNU coreutils versions from the CICD images for comparison:
```
ubuntu-2004: 8.30 (latest)
ubunut-1804: 8.28
macos-latest: 8.32
```
If in the future there's a new ubuntu-latest with GNU coreutils 8.34 installed, the test will still function.
_Maintenance should not be required._

If there's something I didn't think of and hard coding here is indeed bad,an alternative for hard coding `VERSION_EXPECTED` could be to parse e.g.
https://github.com/uutils/coreutils/blob/9d5957f0953694944359c98431055a72fb9e28b1/.github/workflows/GnuTests.yml#L21
But then it must be communicated that changing/removing this file/line can break some tests.

Reply:
I changed them according to your suggestions.